include/nbl/builtin/hlsl/emulated/float64_t_impl.hlsl

```diff
@@ -0,0 +1,316 @@
+        return val;
+
+    int exp = findMSB(val);
+    uint64_t mantissa;
+
+    int shiftCnt = 52 - exp;
+    if (shiftCnt >= 0)
+    {
+        mantissa = val << shiftCnt;
+    }
+    else
+    {
+        const int shiftCntAbs = -shiftCnt;
+        uint64_t roundingBit = 1ull << (shiftCnt - 1);
+        uint64_t stickyBitMask = roundingBit - 1;
+        uint64_t stickyBit = val & stickyBitMask;
+
+        mantissa = val >> shiftCntAbs;
+
+        if ((val & roundingBit) && (!stickyBit))
+        {
+            bool isEven = mantissa & 1;
+            if (!isEven)
+                mantissa++;
+        }
+        else if ((val & roundingBit) && (stickyBit || (mantissa & 1)))
+            val += roundingBit;
+
+        //val += (1ull << (shiftCnt)) - 1;
+        //mantissa = val >> shiftCntAbs;
+
+        if (mantissa & 1ull << 53)
+        {
+            mantissa >>= 1;
+            exp++;
+        }
+    }
+    mantissa &= ieee754::traits<float64_t>::mantissaMask;
+    const uint64_t biasedExp = uint64_t(ieee754::traits<float64_t>::exponentBias + exp) << ieee754::traits<float64_t>::mantissaBitCnt;
+
+    return biasedExp | mantissa;
+};
+
+inline uint64_t reinterpretAsFloat64BitPattern(int64_t val)
+{
+    const uint64_t sign = val & ieee754::traits<float64_t>::signMask;
+    const uint64_t absVal = uint64_t(abs(val));
+    return sign | reinterpretAsFloat64BitPattern(absVal);
+};
+
+NBL_CONSTEXPR_INLINE_FUNC uint64_t flushDenormToZero(uint64_t value)
+{
+    const uint64_t biasBits = value & ieee754::traits<float64_t>::exponentMask;
+    return biasBits ? value : (value & ieee754::traits<float64_t>::signMask);
+}
+
+NBL_CONSTEXPR_INLINE_FUNC uint64_t assembleFloat64(uint64_t signShifted, uint64_t expShifted, uint64_t mantissa)
+{
+    return  signShifted | expShifted | mantissa;
+}
+
+NBL_CONSTEXPR_INLINE_FUNC bool areBothInfinity(uint64_t lhs, uint64_t rhs)
+{
+    lhs &= ~ieee754::traits<float64_t>::signMask;
+    rhs &= ~ieee754::traits<float64_t>::signMask;
+
+    return lhs == rhs && lhs == ieee754::traits<float64_t>::inf;
+}
+
+NBL_CONSTEXPR_INLINE_FUNC bool areBothZero(uint64_t lhs, uint64_t rhs)
+{
+    return !bool((lhs | rhs) << 1);
+}
+
+NBL_CONSTEXPR_INLINE_FUNC bool areBothSameSignZero(uint64_t lhs, uint64_t rhs)
+{
+    return !bool((lhs) << 1) && (lhs == rhs);
+}
+
+template<bool FastMath, typename Op>
+NBL_CONSTEXPR_INLINE_FUNC bool operatorLessAndGreaterCommonImplementation(uint64_t lhs, uint64_t rhs)
+{
+    if (!FastMath)
+    {
+        if (hlsl::isnan<uint64_t>(lhs) || hlsl::isnan<uint64_t>(rhs))
+            return false;
+        if (emulated_float64_t_impl::areBothInfinity(lhs, rhs))
+        {
+            const uint64_t lhsSign = ieee754::extractSignPreserveBitPattern(lhs);
+            const uint64_t rhsSign = ieee754::extractSignPreserveBitPattern(rhs);
+
+            if (lhsSign == rhsSign)
+                return false;
+
+            Op compare;
+            return compare(lhs, rhs);
+        }
```

Comment:
IIRC the IEEE 754 standard correctly, for INF, the exponent is all 1s and mantissa is all 0s (non-zero mantissa would be a NaN)

so you can actually treat Inf as MAX_ABS_VAL+1 for the purposes of comparison, and just delete this whole `if` code block